### ISSUE TITLE
Possible fix for your broken unittests

### ISSUE THREAD

**Issue by tblanarik** [convexengineering/gpkit#1573]
I don't know if you've come across this yet, but I think the following breaking change to `pint` is also responsible for your broken unittests:
- https://github.com/hgrecco/pint/issues/1507

You can be reasonably sure because the last test that succeeds ([794](https://acdl.mit.edu/csi/job/CE_gpkit_Push_unit_tests/794/buildnode=windows10x64,optimizer=cvxopt/console)) shows:
- `Successfully installed pint-0.18`

while the first to break shows:
- `Successfully installed pint-0.19.1`

I haven't had a chance to look around the project much yet (I just started by trying to run the unittests), but it seems like the fix is just to update rtd_requirements.txt to read:

```
pint == 0.18
adce
```

If/when I am confident, I can also submit a PR to that effect.

**Comment by tblanarik** [comment]
What's strange is that the rest of the builds seem to be pinned at `pint==0.9`

eg: https://acdl.mit.edu/csi/view/convex%20engineering/job/CE_gpkit_Push_unit_tests/buildnode=macys_VM,optimizer=cvxopt/lastBuild/console

**Comment by whoburg** [comment]
@tblanarik! Great to hear from you! Thanks for the possible fix, great sleuthing. PR would be great.

@galbramc, the pin to `pint==0.9` is interesting -- are we setting that in our unit testing scripts? All I see in `setup.py` is `pint >= 0.8.1`.

**Comment by tblanarik** [comment]
Yeah, I couldn't locate the pin of `pint==0.9`, and was wondering if it's managed outside of this repo by your Jenkins setup 🤷

**Comment by galbramc** [comment]
Hey guys! Yes the jenkins scripts were setup to install `pint == 0.9`. I've removed that now and Jenkins is installing pint-0.20.1. Hopefully you can resolve the issue now!

**Comment by tblanarik** [comment]
@galbramc - I wonder if that's actually going to break the rest of the tests 😆 . 

Since I don't have access to the scripts, I'm wondering if you'd be willing to try pinning _all_ of the Jenkins scripts to 0.18 and rerunning? 

In my own fork, I ran your unittests using GitHub Actions, Python 3.9 (which I think your Jenkins is set to) and the following requirements:

```python
pint==0.18
adce==1.3.3.2
cvxopt==1.3.0
numpy==1.23.1
plotly==5.9.0
scipy==1.8.1
matplotlib==3.5.2
```

and you can see the tests passing on both Ubuntu and Windows, using cvxopt at least (no mosek)

https://github.com/tblanarik/gpkit/actions


Alternatively, is it reasonable to make the Jenkins a part of the repo? 
There must have been some difference between the scripts since only the `windows10x64` builds failed.

<img width="322" alt="CleanShot 2023-02-24 at 15 09 31@2x" src="https://user-images.githubusercontent.com/1554630/221318534-6cfc619b-19eb-40d5-bbfc-4ece908d7e3c.png">


Finally - perhaps you don't really mind if the unittests are broken and this is just noise to you. I can go away in that case 😆

**Comment by galbramc** [comment]
What is preventing us from fixing gpkit and/or the unit tests so it works with the latest pint?

**Comment by whoburg** [comment]
well, latest pint is currently broken per https://github.com/hgrecco/pint/issues/1507, right?

**Comment by galbramc** [comment]
Are we storing any pickles in the repo? Are pickles supposed to be loadable forever or simply be treated as temporary data? Do we just need to recreate the pickled files for the newer pint?

Btw, shouldn't you be getting ready for a ride on a rocket!

**Comment by whoburg** [comment]
ohhhh, good point. yep that's probably a simple fix.

And yes definitely - but I'm also on a campaign to get inbox zero before launch :)

**Comment by galbramc** [comment]
Inbox zero is far more challenging mission! My girlfriend and I would watch the launch if you had picked a more reasonable time of day :)

**Comment by tblanarik** [comment]
The question of loading pickles forever is where I was going to ask you guys what your approach was. If it's that you want GPKit to just always work with the latest releases of packages,  awesome 👍. But if you wanted to maintain some kind of backwards compatibility for particular reasons, then this might be a more complicated fix.

Since I'm only just exploring this project for the first time, I don't know a lot about how people are using it / what sort of support for older packages you care to provide 😁
